backend/app/services/chat_v2/models/resolver.py

```
@@ -0,0 +1,624 @@
+
+    # Get raw values with defaults
+    api_key = env.get("api_key", "")
+    base_url = env.get("base_url", "https://api.openai.com/v1")
+    model_id = env.get("model_id", "gpt-4")
+    model_type = env.get("model", "openai")
+
+    # Resolve environment variable placeholders in api_key
+    # This handles cases like api_key="${WECODE_API_KEY}"
+    if api_key and "${" in api_key:
+        logger.info(
+            f"[model_resolver] api_key contains placeholder, resolving from env..."
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)